build/xtask/src/dist.rs

```diff
@@ -1306,6 +1313,74 @@ fn update_image_header(
     Ok(false)
 }
 
+/// Checks our dump config:  that if we have a dump agent, it has a task slot
+/// for Jefe (denoting task dump support); that every memory that the dump
+/// agent is using it also being used by Jefe; that if dumps are enabled, the
+/// support has been properly enabled in the kernel.  (Conversely, we assure
+/// that if dump support is enabled, the other components are properly
+/// configured.)  Yes, this is some specific knowledge of the system to encode
+/// here, but we want to turn a preventable, high-consequence run-time error
+/// (namely, Jefe attempting accessing memory that it doesn't have access to
+/// or making a system call that is unsupported) into a compile-time one.
+fn check_dump_config(toml: &Config) -> Result<()> {
+    let dump_support = toml.kernel.features.iter().find(|&f| f == "dump");
+
+    if let Some(task) = toml.tasks.get("dump_agent") {
+        if task.extern_regions.is_empty() {
+            bail!(
+                "dump agent misconfiguration: dump agent is present \
+                but does not have any external regions for dumping"
+            );
+        }
+
+        if task.task_slots.get("jefe").is_none() {
+            bail!(
+                "dump agent misconfiguration: dump agent is present \
+                but has not been configured to depend on jefe"
+            );
+        }
+
+        //
+        // We have a dump agent, and it has a slot for Jefe, denoting that
+        // it is configured for task dumps; now we want to check that Jefe
+        // (1) has the dump feature enabled (2) has extern regions and
+        // (3) uses everything that the dump agent is using.
+        //
+        let jefe = toml.tasks.get("jefe").context("missing jefe?")?;
+
+        if jefe.features.iter().find(|&f| f == "dump").is_none() {
+            bail!(
+                "dump agent/jefe misconfiguration: dump agent depends \
+                on jefe, but jefe does not have the dump feature enabled"
+            );
+        }
+
+        if dump_support.is_none() {
+            bail!(
+                "dump agent is present and system is otherwise configured \
+                for dumping, but kernel does not have the dump feature enabled
+            "
+            );
+        }
+
+        for u in &task.extern_regions {
+            if jefe.extern_regions.iter().find(|&j| j == u).is_none() {
+                bail!(
+                    "dump agent/jefe misconfiguration: dump agent has \
+                    {u} as an extern-region and depends on jefe, but jefe \
+                    does not have {u} as an extern-region"
+                );
+            }
+        }
+    } else {
```

Comment:
Nit: Clippy will probably tell you to flatten this conditional.

Reply:
It does! It also tells me a *lot* of other things on this file -- may be worth making `build` Clippy-clean at some point... ;)

Anyway, will fix this one -- thank you!